packages/cli/core/src/commands/ui/create/vue.spec.ts

```diff
@@ -146,43 +185,120 @@ describe('ui:create:vue', () => {
-      );
+  describe('when a file with the projectName exists', () => {
+    beforeEach(() => {
+      const mockedIsDir = jest.fn();
+      mockedIsDir.mockReturnValue(false);
+      mockedStatSync.mockReturnValue({
+        isDirectory: mockedIsDir,
+      } as unknown as Stats);
+    });
+    test
+      .stdout()
+      .stderr()
+      .command(['ui:create:vue', 'myapp'])
+      .catch((err) => {
+        expect(formatAbsolutePath(err.message)).toMatchSnapshot();
+      })
+      .it('should exit with an error', () => {
+        expect(mockedSpawnProcess).not.toBeCalled();
+        expect(mockedStatSync).toBeCalledWith(
+          expect.stringMatching(fooBarDirectoryMatcher),
+          {
+            throwIfNoEntry: false,
+          }
+        );
+      });
+  });
+
+  describe('when a directory with the projectName exists', () => {
+    beforeEach(() => {
+      const mockedIsDir = jest.fn();
+      mockedIsDir.mockReturnValue(true);
+      mockedStatSync.mockReturnValue({
+        isDirectory: mockedIsDir,
+      } as unknown as Stats);
     });
+    describe('when the directory is not empty', () => {
+      beforeEach(() => {
+        mockedReadDirSync.mockReturnValue([{} as unknown as Dirent]);
+      });
+      test
+        .stdout()
+        .stderr()
+        .command(['ui:create:vue', 'myapp'])
+        .catch((err) => {
+          expect(formatAbsolutePath(err.message)).toMatchSnapshot();
+        })
+        .it('should exit with an error', () => {
+          expect(mockedSpawnProcess).not.toBeCalled();
+          expect(mockedStatSync).toBeCalledWith(
+            expect.stringMatching(fooBarDirectoryMatcher),
+            {
+              throwIfNoEntry: false,
+            }
+          );
+          expect(mockedReadDirSync).toBeCalledWith(
+            expect.stringMatching(fooBarDirectoryMatcher),
+            {
+              withFileTypes: true,
+            }
+          );
+        });
+    });
+    describe('when the directory is empty', () => {
+      beforeEach(() => {
+        mockedReadDirSync.mockReturnValue([]);
+      });
+      test
+        .stdout()
+        .stderr()
+        .command(['ui:create:vue', 'myapp'])
+        .it('should use the dir and init the project', () => {
+          expect(mockedStatSync).toBeCalledWith(
+            expect.stringMatching(fooBarDirectoryMatcher),
+            {
+              throwIfNoEntry: false,
+            }
+          );
+          expect(mockedReadDirSync).toBeCalledWith(
+            expect.stringMatching(fooBarDirectoryMatcher),
+            {
+              withFileTypes: true,
+            }
+          );
+          expect(mockedSpawnProcess).toBeCalledTimes(2);
+          expect(mockedSpawnProcess.mock.calls[0]).toMatchSnapshot();
+          expect(mockedSpawnProcess.mock.calls[1]).toMatchSnapshot();
+          expect(mockedWriteFileSync).toBeCalledTimes(1);
+          expect(mockedWriteFileSync.mock.calls[0]).toMatchSnapshot();
```

Comment:
Seems like there are too many expects in the same test.

Reply:
Split